brainframe_qt/api_utils/__init__.py

```diff
@@ -1,5 +1,22 @@
+import typing
+
+from PyQt5.QtCore import QObject
 from brainframe.api import BrainFrameAPI
-from .streaming import StreamManager, StreamManagerAPI
 
-# API instance that is later monkeypatched to be a singleton
-api: BrainFrameAPI = StreamManagerAPI()
+# Singleton API instance
+api = BrainFrameAPI()
+
+# Must come after api import
+from .streaming import StreamManager
+
+# Set using init_stream_manager before use
+_stream_manager: StreamManager = typing.cast(StreamManager, None)
+
+
+def init_stream_manager(*, parent: QObject) -> None:
+    global _stream_manager
```

Comment:
Should you check if the stream_manager already has a value? I assume you don't want this function being called twice and re-creating a StreamManager each time.

Reply:
Good idea. Will add that